File: practice/algorithms/implementation/beautiful_days_at_the_movies/BeautifulDaysAtTheMovies.java
import java.util.*;

public class Solution {

    public static void main(String[] args) {
        Scanner stdin = new Scanner(System.in);
        int start = stdin.nextInt();
        int end = stdin.nextInt();
        int k = stdin.nextInt();
        int count = 0;
        for(int i = start; i <= end; i++) {
            String s = String.valueOf(i);
            String reversed = new StringBuilder(s).reverse().toString();
            int diff = Math.abs(Integer.parseInt(s) - Integer.parseInt(reversed));
            if(diff % k == 0) {
                count++;
            }
        }
        System.out.println(count);
        stdin.close();
    }
}

File: practice/algorithms/implementation/beautiful_days_at_the_movies/BeautifulDaysAtTheMovies.js
'use strict';

const reverse = (s) => s.split("").reverse().join("");

const processData = input => {
    const [start, end, d] = input.split(' ').map(i => parseInt(i));
    let count = 0;
    for(let i = start; i <= end; i++) {
        const s = i.toString();
        const reversed = reverse(s);
        const diff = Math.abs(parseInt(s) - parseInt(reversed));
        if(diff % d == 0) {
            count++;
        }
    }
    console.log(count);
};

process.stdin.resume();
process.stdin.setEncoding("ascii");

let _input = "";
process.stdin.on("data", input => _input += input);
process.stdin.on("end", () => processData(_input));